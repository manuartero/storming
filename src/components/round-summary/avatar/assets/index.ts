import enemy1 from "./enemy1.svg";
import enemy2 from "./enemy2.svg";
import enemy3 from "./enemy3.svg";
import player from "./player.svg";

export const avatarAssets = {
  player,
  enemy1,
  enemy2,
  enemy3,
};
