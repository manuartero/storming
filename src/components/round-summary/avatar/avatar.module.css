.avatar {
  --player-color: var(--dark-grey);

  position: relative;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;

  width: 6vh;
  height: 6vh;

  pointer-events: none;
}

/*
.avatar.player {
  --player-color: var(--player-primary);
}

.avatar.enemy1 {
  --player-color: var(--enemy1-primary);
}

.avatar.enemy2 {
  --player-color: var(--enemy2-primary);
}

.avatar.enemy3 {
  --player-color: var(--enemy3-primary);
}
*/
