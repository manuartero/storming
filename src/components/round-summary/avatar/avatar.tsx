import c from "classnames";
import { avatarAssets } from "./assets";

import styles from "./avatar.module.css";

type Props = {
  player?: PlayerType;
};

export function Avatar({ player = "player" }: Props) {
  const icon = avatarAssets[player];

  return (
    <div
      role="img"
      className={c(styles.avatar, styles[player])}
      aria-roledescription="game avatar"
      aria-label={`${player} avatar`}
      style={{ backgroundImage: `url(${icon})` }}
    />
  );
}
