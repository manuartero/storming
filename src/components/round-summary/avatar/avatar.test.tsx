import { render, screen } from "@testing-library/react";
import { Avatar } from "./avatar";

describe("<Avatar />", () => {
  test('render: "player" avatar', () => {
    render(<Avatar />);
    const playerAvatar = screen.getByRole("img", { name: "player avatar" });
    expect(playerAvatar).toMatchSnapshot();
  });

  test('render: "enemy1" avatar', () => {
    render(<Avatar player="enemy1" />);
    const enemy1Avatar = screen.getByRole("img", { name: "enemy1 avatar" });
    expect(enemy1Avatar).toBeInTheDocument();
  });
});
