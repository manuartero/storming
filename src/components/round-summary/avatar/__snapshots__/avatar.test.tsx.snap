// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Avatar /> render: "player" avatar 1`] = `
<div
  aria-label="player avatar"
  aria-roledescription="game avatar"
  class="avatar player"
  role="img"
/>
`;
